### PR TITLE
Switch to BinaryProvider

### LINKED ISSUE #54: Missing make, gcc dependencies
I'm building a script for installing my software in a production environment on a VM. I found that if I copy my REQUIRE file over there, everything works fine, except for RMath.jl, which is missing `make` and `gcc`. I could install them as part of my script

```
apt-get -y install make
apt-get -y install gcc
```

but it would be cleaner if it was handled as part of RMath's dependencies.

### PR DESCRIPTION
**Pull request by andreasnoack**
Supersedes #48. All supported platforms are built for but obviously not tested.

Fixes #54

### CODE REVIEW

**Comment by andreasnoack** [comment]
Looks like this is working. the failures on Linux is because `gamma` has been moved out of base.

File: REQUIRE [hunk 1]
```diff
@@ -1,3 +1,3 @@
 julia 0.6
-BinDeps 0.6.0
+BinaryProvider 0.3.2
```
Comment (tkelman) [inline REQUIRE]: what's necessary from 0.3.2 that wasn't in 0.3.0 (or earlier?)

**Comment by ararslan** [comment]
Confirmed to work and pass all tests on FreeBSD with Julia 0.6.

File: REQUIRE (hunk 1 above)
Comment (andreasnoack) [reply REQUIRE]: There have been bug fixes added lately and it's the only version which has been tested so it's the only version known to work. Feel free to test prior versions and submit a PR if you can verify that it works for prior versions.

Comment (tkelman) [reply REQUIRE]: are the bug fixes needed here? that's not what minimum versions are for. they are lowest version that should work based on features being used, not version that has been tested.

Comment (tkelman) [reply REQUIRE]: Hello? You shouldn't release things that are just wrong.

**Pull request merged**